lib/compat/wordpress-6.5/fonts/fonts.php

```diff
@@ -1,485 +0,0 @@
-		 *     @type string       $url     Base URL and subdirectory or absolute URL to the fonts upload directory.
-		 *     @type string       $subdir  Subdirectory
-		 *     @type string       $basedir Path without subdir.
-		 *     @type string       $baseurl URL path without subdir.
-		 *     @type string|false $error   False or error message.
-		 * }
-		 */
-		return apply_filters( 'font_dir', $font_dir );
-	}
-}
-
-// @core-merge: Filters should go in `src/wp-includes/default-filters.php`,
-// functions in a general file for font library.
-if ( ! function_exists( '_wp_after_delete_font_family' ) ) {
-	/**
-	 * Deletes child font faces when a font family is deleted.
-	 *
-	 * @access private
-	 * @since 6.5.0
-	 *
-	 * @param int     $post_id Post ID.
-	 * @param WP_Post $post    Post object.
-	 */
-	function _wp_after_delete_font_family( $post_id, $post ) {
-		if ( 'wp_font_family' !== $post->post_type ) {
-			return;
-		}
-
-		$font_faces = get_children(
-			array(
-				'post_parent' => $post_id,
-				'post_type'   => 'wp_font_face',
-			)
-		);
-
-		foreach ( $font_faces as $font_face ) {
-			wp_delete_post( $font_face->ID, true );
-		}
-	}
-	add_action( 'deleted_post', '_wp_after_delete_font_family', 10, 2 );
-}
-
-if ( ! function_exists( '_wp_before_delete_font_face' ) ) {
-	/**
-	 * Deletes associated font files when a font face is deleted.
-	 *
-	 * @access private
-	 * @since 6.5.0
-	 *
-	 * @param int     $post_id Post ID.
-	 * @param WP_Post $post    Post object.
-	 */
-	function _wp_before_delete_font_face( $post_id, $post ) {
-		if ( 'wp_font_face' !== $post->post_type ) {
-			return;
-		}
-
-		$font_files = get_post_meta( $post_id, '_wp_font_face_file', false );
-		$font_dir   = untrailingslashit( wp_get_font_dir()['basedir'] );
-
-		foreach ( $font_files as $font_file ) {
-			wp_delete_file( $font_dir . '/' . $font_file );
-		}
-	}
-	add_action( 'before_delete_post', '_wp_before_delete_font_face', 10, 2 );
-}
-
-// @core-merge: Do not merge this function, it is for deleting fonts from the wp-content/fonts directory only used in Gutenberg.
-/**
- * Deletes associated font files from wp-content/fonts, when a font face is deleted.
- *
- * @param int     $post_id Post ID.
- * @param WP_Post $post    Post object.
- */
-function gutenberg_before_delete_font_face( $post_id, $post ) {
-	if ( 'wp_font_face' !== $post->post_type ) {
-		return;
-	}
-
-	$font_files = get_post_meta( $post_id, '_wp_font_face_file', false );
-
-	if ( empty( $font_files ) ) {
-		return;
-	}
-
-	$site_path = '';
-	if ( is_multisite() && ! ( is_main_network() && is_main_site() ) ) {
-		$site_path = '/sites/' . get_current_blog_id();
-	}
-
-	$font_dir = path_join( WP_CONTENT_DIR, 'fonts' ) . $site_path;
-
-	foreach ( $font_files as $font_file ) {
-		$font_path = $font_dir . '/' . $font_file;
-
-		if ( file_exists( $font_path ) ) {
-			wp_delete_file( $font_path );
-		}
-	}
-}
-add_action( 'before_delete_post', 'gutenberg_before_delete_font_face', 10, 2 );
```

Comment:
As I [described here](https://github.com/WordPress/gutenberg/pull/57688#issuecomment-2259037546), I think we should leave this code in, for now. Perhaps `lib/compat/plugin/fonts.php` would be a good place for it?

Reply:
Makes sense. Thank you, @creativecoder!